subiquity/server/tests/test_server.py

```diff
@@ -232,6 +262,70 @@ async def test_autoinstall_validation__no_error_report(self):
         error = NonReportableError.from_exception(exception)
         self.assertEqual(error, self.server.nonreportable_error)
 
+    @patch("subiquity.server.server.log")
+    async def test_autoinstall_validation__strict_top_level_keys_warn(self, log_mock):
+        """Test strict top-level key enforcement warnings in v1"""
+
+        bad_ai_data = {
+            "version": 1,
+            "interactive-sections": ["identity"],
+            "apt": "Invalid but deferred",
+            "literally-anything": "lmao",
+        }
+
+        good, bad = self.load_config_and_controllers(bad_ai_data)
+
+        # OK in Version 1 but ensure warnings and stripped config
+        self.server.validate_autoinstall()
+        log_mock.warning.assert_called()
+        log_mock.error.assert_not_called()
+        self.assertEqual(self.server.autoinstall_config, good)
```

Comment:
minor: maybe assert empty here on bad, more for documentation sake than anything else

Reply:
Not quite empty in this case but I get what you mean. I added the check for the bad config.